xdgdesktopfile.cpp

```diff
@@ -462,7 +462,7 @@ bool XdgDesktopFileData::startByDBus(const QStringList& urls) const
         return false;
     }
     QDBusInterface app(f.completeBaseName(), path, QLatin1String("org.freedesktop.Application"));
-    if (!app.isValid())
+    if (app.lastError().isValid())
```

Comment:
I'd write `if (!app.isValid() || app.lastError().isValid())` just for cautiosness.